### PR TITLE
dev-python/{pytools,pyopencl}: version bump

### PR DESCRIPTION
**Pull request by t0b3**
Closes: https://bugs.gentoo.org/show_bug.cgi?id=712672
Closes: https://bugs.gentoo.org/show_bug.cgi?id=712680
Package-Manager: Portage-2.3.89, Repoman-2.3.20
Signed-off-by: t0b3 <thomas.bettler@gmail.com>

### CODE REVIEW

**Comment by t0b3** [comment]
@jonasstein: any comments on QA?
@mgorny: I'd like to keep dev-python/pyopencl in portage ([bug 711808](https://bugs.gentoo.org/711808)) - is there anything restraining to update & drop p.mask?

File: dev-python/pyopencl/pyopencl-2019.1.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	>=virtual/opencl-2"
+DEPEND="${RDEPEND}"
+
+src_configure() {
+	local myconf=()
+	if use opengl; then
+		myconf+=(--cl-enable-gl)
+	fi
+
+	"${PYTHON}" configure.py \
```
Comment (mgorny) [inline dev-python/pyopencl/pyopencl-2019.1.2.ebuild]: EPYTHON

File: dev-python/pyopencl/pyopencl-2019.1.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		--boost-compiler=gcc \
+		--boost-python-libname=boost_python-${PYTHON_ABI}-mt \
+		--no-use-shipped-boost \
+		--cl-pretend-version=1.2 \
+		"${myconf[@]}"
+}
+
+python_install_all() {
+	if use examples; then
+		local EXAMPLES=( examples/. )
```
Comment (mgorny) [inline dev-python/pyopencl/pyopencl-2019.1.2.ebuild]: This is banned.

File: dev-python/pyopencl/pyopencl-2019.1.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	>=dev-python/decorator-3.2.0[${PYTHON_USEDEP}]
+	dev-python/mako[${PYTHON_USEDEP}]
+	dev-python/numpy[${PYTHON_USEDEP}]
+	>=dev-python/pytools-2017.5[${PYTHON_USEDEP}]
+	dev-python/pybind11[${PYTHON_USEDEP}]
+	>=dev-python/six-1.9.0[${PYTHON_USEDEP}]
+	>=virtual/opencl-2"
+DEPEND="${RDEPEND}"
+
+src_configure() {
```
Comment (mgorny) [inline dev-python/pyopencl/pyopencl-2019.1.2.ebuild]: python_configure_all

File: dev-python/pytools/pytools-2020.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	>=dev-python/decorator-3.2.0[${PYTHON_USEDEP}]
+	>=dev-python/numpy-1.6.0[${PYTHON_USEDEP}]
+	>=dev-python/six-1.8.0[${PYTHON_USEDEP}]
+"
+DEPEND="
+	>=dev-python/setuptools-0.7.2[${PYTHON_USEDEP}]
+	test? (
+		${RDEPEND}
+		dev-python/pytest[${PYTHON_USEDEP}]
+		dev-python/pytest-runner[${PYTHON_USEDEP}]
```
Comment (mgorny) [inline dev-python/pytools/pytools-2020.1.ebuild]: Kill this.

File: dev-python/pytools/pytools-2020.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+	)
+"
+
+python_test() {
```
Comment (mgorny) [inline dev-python/pytools/pytools-2020.1.ebuild]: Use distutils_enable_tests.

File: dev-python/pyopencl/pyopencl-2019.1.2.ebuild [hunk 6]
```diff
@@ -0,0 +1,53 @@
+# Copyright 1999-2020 Gentoo Authors
```
Comment (mgorny) [inline dev-python/pyopencl/pyopencl-2019.1.2.ebuild]: Any chance for tests here?

File: dev-python/pyopencl/pyopencl-2019.1.2.ebuild (hunk 1 above)
Comment (t0b3) [reply dev-python/pyopencl/pyopencl-2019.1.2.ebuild]: THX, switched to ${EPYTHON}

File: dev-python/pyopencl/pyopencl-2019.1.2.ebuild (hunk 2 above)
Comment (t0b3) [reply dev-python/pyopencl/pyopencl-2019.1.2.ebuild]: THX, I reimplemented examples like here https://dev.gentoo.org/~mgorny/python-guide/distutils.html#python-install

File: dev-python/pyopencl/pyopencl-2019.1.2.ebuild (hunk 3 above)
Comment (t0b3) [reply dev-python/pyopencl/pyopencl-2019.1.2.ebuild]: THX, changed to python_configure_all

File: dev-python/pytools/pytools-2020.1.ebuild (hunk 4 above)
Comment (t0b3) [reply dev-python/pytools/pytools-2020.1.ebuild]: dropped this in favor of _distutils_enable_tests pytest_

File: dev-python/pytools/pytools-2020.1.ebuild (hunk 5 above)
Comment (t0b3) [reply dev-python/pytools/pytools-2020.1.ebuild]: dropped this in favor of _distutils_enable_tests pytest_

File: dev-python/pyopencl/pyopencl-2019.1.2.ebuild (hunk 6 above)
Comment (t0b3) [reply dev-python/pyopencl/pyopencl-2019.1.2.ebuild]: added tests

File: dev-python/pytools/pytools-2020.1.ebuild (hunk 4 above)
Comment (mgorny) [reply dev-python/pytools/pytools-2020.1.ebuild]: Actually, could you 'rebase' this file on the current version? It feels silly to copy the old version, then redo part of the changes that were done in it.

Comment (t0b3) [reply dev-python/pytools/pytools-2020.1.ebuild]: What I understand is that you are not happy yet with this contribution. However I missed somehow what you would prefer instead. 
I understood you probably want one squashed/atomic commit (per package). Was this your point? Thanks for clarifying if you meant something different.

Comment (t0b3) [reply dev-python/pytools/pytools-2020.1.ebuild]: one squashed/atomic commit (per package) is in tree now

**Comment by t0b3** [comment]
@mgorny I realized that you probably referred to the changes introduced meanwhile by https://github.com/gentoo/gentoo/commit/8c912d54eaaf4e7d289f40683ae0bb6b340bf969. 
That's included now.

**Comment by mgorny** [comment]
d_e_t is the preferred method nowadays.

File: dev-python/pyopencl/pyopencl-2019.1.2.ebuild [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+		myconf+=(--cl-enable-gl)
+	fi
+
+	"${EPYTHON}" configure.py \
+		--cl-pretend-version=1.2 \
+		"${myconf[@]}"
+}
+
+python_test() {
+	cd "${S}/test"
```
Comment (mgorny) [inline dev-python/pyopencl/pyopencl-2019.1.2.ebuild]: ||die

File: dev-python/pyopencl/pyopencl-2019.1.2.ebuild [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+	pytest -v || die "Testsuite failed under ${EPYTHON}"
```
Comment (mgorny) [inline dev-python/pyopencl/pyopencl-2019.1.2.ebuild]: -vv

**Comment by t0b3** [comment]
@mgorny PR is changed as requested.
```
 python_test() {
-       cd "${S}/test"
-       pytest -v || die "Testsuite failed under ${EPYTHON}"
+       cd "${S}/test" | die
+       pytest -vv || die "Testsuite failed under ${EPYTHON}"
 }

```
In case you see more things to change, just tell me...

File: dev-python/pyopencl/pyopencl-2019.1.2.ebuild (hunk 7 above)
Comment (t0b3) [reply dev-python/pyopencl/pyopencl-2019.1.2.ebuild]: done

File: dev-python/pyopencl/pyopencl-2019.1.2.ebuild (hunk 8 above)
Comment (t0b3) [reply dev-python/pyopencl/pyopencl-2019.1.2.ebuild]: done

**Comment by mgorny** [comment]
I've fixed the commit message for you and merged it.

**Comment by t0b3** [comment]
Thank you

**Pull request closed without merging**